solution.cpp

```diff
@@ -23,112 +23,203 @@ using std::ifstream;
+* - `infilename`: A string representing the file path to read from.
+* - `outfilename`: A string representing the file path to write to.
+*
+* Returns:
+* - 0 on success, or -1 on error.
+*/
+int write_composites(const int size,
+	bool primes[],
+	const string& infilename,
+	const string& outfilename);
 
 // ----------------------------------------------------------------------------
 
 int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
-        return 1;  // error
-    }
-
-    bool primes[max+1];
-    sieve(max+1, primes);
-
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
-    }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
-    }
-
-    return 0;  // success
+	int max = find_max("input.txt");
+	if (max == -1) {
+		cout << "ERROR in `find_max()`" << endl;
+		return 1;  // error
+	}
+	
+	bool primes[max + 1];				// This particular part gives an error in Visual Studio, but if I hard code it (to 11 in the example case), it works, no idea how to fix
+	sieve(max + 1, primes);
+
+	
+
+	//for (int value = 0; value < 11; value++) {
+	//	cout << primes[value] << endl;
+	//}
+	//cin.get();
+
+
+
+	int ret;  // for storing return values, to check for error codes
+	//
+	ret = write_primes(max + 1, primes, "input.txt", "primes.txt");
+	if (ret == -1) {
+		cout << "ERROR in `write_primes()`" << endl;
+		return 1;  // error
+	}
+	//
+	ret = write_composites(max + 1, primes, "input.txt", "composites.txt");
+	if (ret == -1) {
+		cout << "ERROR in `write_composites()`" << endl;
+		return 1;  // error
+	}
+
+	return 0;  // success
 }
 
 // ----------------------------------------------------------------------------
 
 // TODO: write prototyped functions
 
+
+
+int find_max(const string& infilename) {			//The find_max definition
+	ifstream inputFile;
+	inputFile.open(infilename);
+	int inputnumber;
+	int theMaximum = 0;
+	while (inputFile >> inputnumber) {
+		if (inputnumber > theMaximum) {
+			theMaximum = inputnumber;
+		}
+	}
+	inputFile.close();
+	return theMaximum;
+}
+
+void sieve(const int size, bool primes[]) {			//The sieve definition
+	int count = 0;
+	int divisor;
+	for (count; count < size; count++) {
+		if (count == 1) {
+			primes[count] = 0;
+		}
+		if (count == 2) {
+			primes[count] = 1;
+		}
+		for (divisor = 2; divisor < count; divisor++) {
+			if (count % divisor == 0) {
+				primes[count] = 0;
+				break;
+			}
+			else {
+				primes[count] = 1;
+			}
+		}
+	}
+}
+
+
+
+
+int write_primes(const int size,					//The write_primes definition
+	bool primes[],
+	const string& infilename,
+	const string& outfilename) {
```

Comment:
please be careful with formatting like this -- on first glance it's hard to tell that these lines are part of the function's parameter list